src/app/doctor.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { HttpClient} from '@angular/common/http';
+import { Doctor } from './Doctor';
+import { Observable} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DoctorService {
+
+  private addUrl = "http://localhost:8080/doctor/auth";
```

Comment:
Do not hard code external URLs like this. Use configuration JSON or any other better approach.

Reply:
done sir